### PR TITLE
[Site design revamp] Open preview on thumbnail tap

### LINKED ISSUE #16407: [Site Design Revamp] Main View - Tapping a design launches preview
## Description

Related to the changes included in [#16392](https://github.com/wordpress-mobile/WordPress-Android/issues/16392), tapping on a design should now open the preview directly. This will allow us to change this functionality before a [carousel view](https://github.com/wordpress-mobile/WordPress-Android/issues/16395) is created, in case it should be moved to a later phase.

**Before:**
- Tapping a design selected it and added a check mark icon
- A selected design would shown the bottom actions buttons (Preview, Choose)
- Tapping "Preview" would launch the design preview

**After:**
- Tapping a design launches the design preview

PRs should target the `feature/site-design-revamp` feature branch.

### PR DESCRIPTION
**Pull request by mkevins**
Fixes #16407 

### Description

This PR wires up the thumbnails in the theme picker to open the preview when tapped. The approach relies on some existing logic, to avoid deeply bifurcating the existing shared implementation with the page picker. Also, since the underlying picker can still be visible on wider screens (beneath the preview modal), the selected state for the thumbnail is updated when tapped, and deselected when the preview is dismissed.

##### Remaining tasks:
- [x] Add automated tests
- [x] Accessibility considerations

#### Portrait

1. Start the site creation flow
2. Pick an intent (or skip)
3. Enter a site name (or skip)
4. Expect to land on the theme selection screen
5. Tap a theme
6. Expect the preview to open for that theme
7. Tap back
8. Expect to land on the theme selection screen
   * Expect the theme to become deselected

#### Landscape (or tablet)

1. Start the site creation flow
2. Pick an intent (or skip)
3. Enter a site name (or skip)
4. Expect to land on the theme selection screen
5. Tap a theme
6. Expect the preview to open for that theme
   * Expect that the tapped thumbnail (if visible) is in a selected state (picking the left-most one is helpful here)
7. Tap back
8. Expect to land on the theme selection screen
   * Expect the theme to become deselected

#### Accessibility

1. Turn on talkback
2. Follow steps 1-5 from the Portrait test above
   * Expect that tapping a theme announces "Preview theme [site title]" for each thumbnail
   * It is helpful to scroll down to the Professional or Business themes, such as Easley, Vesta, Alves, Barnsbury, etc. since many of the others seem to have "Site title" as their title. This is an issue that can be tackled on the backend.

### Page Picker

1. Navigate to the page list and tap the FAB to start the create a new page flow
4. Tap a page layout thumbnail to select it, and tap again to deselected it
   * Expect it to become selected (with a check mark and border)
   * Expect the bottom buttons to change from "Create blank page" to "Preview" and "Create page"
5. Tap "Preview"
   * Expect preview to load
6. Tap back button
   * Expect to return to the page picker screen with the tapped thumbnail in the selected state

## Regression Notes
1. Potential unintended areas of impact
Modal layout picker

10. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing to ensure the UI for the page picker has not changed.

11. What automated tests I added (or what prevented me from doing so)
Tests have been updated for the new ui. Also, there was one test which would have been desired, but was a behavior that exists outside of the SUT, [described here](https://github.com/wordpress-mobile/WordPress-Android/pull/16480/commits/8275b865426fcb82e6bb5ad19fb927ead899d118).

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by antonis** [comment]
I briefly tested the functionality and it worked as expected. The approach chosen on the coding side also LGTM 👍 
I'll be glad to revisit once the PR is ready for review :)

**Comment by mkevins** [comment]
Thank you for testing this Antonis, and having a look at the approach!

I've added a test here https://github.com/wordpress-mobile/WordPress-Android/pull/16480/commits/949bc2ac86c2bba481d9016996a98fcf697e707c#diff-fc69a3faa397c5cc169551db1b81be29a1bd59a2724c53fee8354588c8443a0dR158 which "mirrors" the assertion here https://github.com/wordpress-mobile/WordPress-Android/pull/16480/commits/949bc2ac86c2bba481d9016996a98fcf697e707c#diff-fc69a3faa397c5cc169551db1b81be29a1bd59a2724c53fee8354588c8443a0dL154, but I had a second thought about this. It seems the desired behavior was to prevent selection of thumbnails for which the MShot image has not yet loaded. This makes sense to me, but the reason for my hesitation is that this means we make the user wait longer than necessary if they are utilizing the talkback feature (i.e. in such a case, they would not necessarily benefit from waiting for the image to load, and could proceed with the content description, which is already available). Perhaps this is an inevitable trade-off, and I've provided the changes to keep it consistent with what we've done in the past - just wanted to mention these thoughts, since I'd been recently testing the accessibility of this flow.

As a separate issue, I observed that many site titles are simply "Site title", and are used as a content description for describing the thumbnails. This comes from the back-end, so I think we can fix this by updating the titles of the demo sites (perhaps a simple solution is to just match the slug). I'll draft an issue for this.

This is ready for review now. :+1:

File: WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerTracker.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package org.wordpress.android.ui.layoutpicker
 interface LayoutPickerTracker {
     fun trackPreviewModeChanged(mode: String)
 
+    // NOOP implementation provided to make override optional in subclasses
```
Comment (mkevins) [inline WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerTracker.kt:6]: Sneaking this comment in here to close the loop [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16468#discussion_r864490457). :smile:

Comment (ovitrif) [reply WordPress/src/main/java/org/wordpress/android/ui/layoutpicker/LayoutPickerTracker.kt:6]: Awesome, thank you for identifying this improvement and addressing it here 🙇

**Review by antonis — APPROVED** [review]
Great work @mkevins ⭐ 
I've tested the implementation on a Pixel 5 (Android 12) and everything worked as expected. The code changes also look good 🎉 
Thank you for also handling accessibility and adding unit tests 🙇 

> As a separate issue, I observed that many site titles are simply "Site title", and are used as a content description for describing the thumbnails. This comes from the back-end, so I think we can fix this by updating the titles of the demo sites (perhaps a simple solution is to just match the slug). I'll draft an issue for this.

That sounds good 👍

**Review by ovitrif — COMMENTED** [review]
Tested and everything went well in my tests too 🎉 
Agreeing also with the changes 🙇‍♂️

**Pull request merged**